src/weatherDataSource.js

```diff
@@ -0,0 +1,42 @@
+export default class WeatherDataSource {
+    constructor() {
+        this.apiKey = '84af29f03ef35bb4dd7f1e9b0ad575e2';
+    }  
+    getweather(city) { 
+        fetch('https://api.openweathermap.org/data/2.5/weather?q=' + city + '&appid=' + this.apiKey)  
+            .then(function(resp) { 
+                return resp.json() })
+            .then(function(data) {
+                console.log(data);
+                showWeather(data);
+            })
+            .catch(function(err) {
+                console.log(err);
+            });
+    }
```

Comment:
This function - that should be getting weather data and returning it - STILL returns nothing. Instead it interacts with the DOM.

Even IF you wanted to avoid returning promises and async/await constructions, the function invoked here should be a callback (be it supplied to the constructor or the method itself), definitely not a random function hard-coded into the file and not even part of the class...

Reply:
I added return